internal/statemachine/classic_states.go

```diff
@@ -581,7 +581,7 @@ func (stateMachine *StateMachine) installPackages() error {
 	var installPackagesCmds []*exec.Cmd
 
 	// mount some necessary partitions from the host in the chroot
-	mounts := []string{"/dev", "/proc", "/sys"}
+	mounts := []string{"/dev", "/proc", "/sys", "/run"}
```

Comment:
I'd like more context here as well. I'm also interested in how livecd-rootfs and live-build deal with this. Since the current ubuntu-base tarballs do not have those things in `/run`, so... why? What does live-build do differently? We shouldn't aim for 1:1 behaviour, as we want to do better, but it's helpful to understand older tooling when creating the next generation.

I'm interested in live-build behaviour here as we need to make sure that the right things happen for all builds - tarballs and preinstalled. A quick check on livecd-rootfs doesn't give me much info, but I might try digging deeper tomorrow. Might be something live-build itself does.

Reply:
Here's a list of stuff showing up in /run:

```
drwxr-xr-x root/root             0 2023-01-24 13:35 ./run/
drwxr-xr-x root/root             0 2023-01-24 13:35 ./run/log/
lrwxrwxrwx root/root             0 2023-01-24 13:35 ./run/shm -> /dev/shm
-rw-r--r-- root/root            41 2023-01-24 13:35 ./run/reboot-required.pkgs
drwxr-xr-x root/root             0 2023-01-24 13:35 ./run/systemd/
drwxr-xr-x systemd-resolve/systemd-network 0 2023-01-24 13:35 ./run/systemd/netif/
drwxr-xr-x systemd-resolve/systemd-network 0 2023-01-24 13:35 ./run/systemd/netif/lldp/
drwxr-xr-x systemd-resolve/systemd-network 0 2023-01-24 13:35 ./run/systemd/netif/leases/
drwxr-xr-x systemd-resolve/systemd-network 0 2023-01-24 13:35 ./run/systemd/netif/links/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/seats/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/users/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/sessions/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/ask-password/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/shutdown/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/resolve/
-rw-r--r-- root/root                     935 2023-01-24 13:35 ./run/systemd/resolve/stub-resolv.conf
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/systemd/machines/
-rw-r--r-- root/root                      32 2023-01-24 13:35 ./run/reboot-required
drwxrwxrwt root/root                       0 2023-01-24 13:35 ./run/lock/
drwx------ root/root                       0 2023-01-24 13:35 ./run/lock/lvm/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/lock/subsys/
drwx------ root/root                       0 2023-01-24 13:35 ./run/lvm/
drwxr-xr-x root/root                       0 2023-01-24 13:33 ./run/mount/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/user/
drwxrwxrwt root/utmp                       0 2023-01-24 13:35 ./run/screen/
drwxr-xr-x root/root                       0 2023-01-24 13:35 ./run/sendsigs.omit.d/
```

It's also worth noting that adding /run to the list of mountpoints here fixed the following bug: https://bugs.launchpad.net/ubuntu-image/+bug/2000883

I haven't had time to look into why that bug is fixed by adding /run to the list of mountpoints.